### PR TITLE
[WIP] feat(core): convert components to typescript

### PR DESCRIPTION
**Pull request by redallen**
**What**: Start undergoing the effort to convert some components to typescript.

- [x] AboutModal
- [x] Alert
- [ ] ApplicationLauncher
- [x] Avatar (tweaked)
- [ ] Backdrop
- [ ] BackgroundImage
- [ ] Badge
- [ ] Brand
- [ ] Breadcrumb
- [ ] Button
- [ ] Card
- [ ] Checkbox
- [ ] ChipGroup
- [ ] DataList
- [ ] Dropdown
- [ ] EmptyState
- [ ] Form
- [ ] FormSelect
- [ ] InputGroup
- [ ] Label
- [ ] List
- [ ] LoginPage
- [ ] Modal
- [ ] Nav
- [ ] Page
- [ ] Popover
- [ ] Progress
- [ ] Radio
- [ ] Select
- [ ] Switch
- [ ] Tabs
- [ ] Text
- [ ] TextArea
- [ ] TextInput
- [ ] Title
- [ ] Tooltip

**Additional issues**: Some components are functional, and some are stateful. Functional ones are trivial to convert and usually only require fixing some type errors regarding properties. Stateful ones are not, often requiring lifecycle fixes.

### CODE REVIEW

File: packages/patternfly-4/react-core/src/components/Badge/Badge.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+export interface BadgeProps extends DetailedHTMLProps<HTMLAttributes<HTMLSpanElement>, HTMLSpanElement> {
+  /**  Adds styling to the badge to indicate it has been read */
+  isRead: boolean;
+  /** content rendered inside the Badge */
+  children: ReactNode;
+  /** additional classes added to the Badge */
+  className?: string;
+}
+
+const Badge: React.FunctionComponent<BadgeProps> = (props) => (
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/Badge/Badge.tsx]: Should destructure and then use props directly
```suggestion
const Badge: React.FunctionComponent<BadgeProps> = ({ isRead, className, children, ...props }) => (
```

Comment (dlabaj) [reply packages/patternfly-4/react-core/src/components/Badge/Badge.tsx]: @redallen This should work for typescript ({isRead, className, children, ...props}: BadgeProps).  I think the avatar one should be updated to this as well.

Comment (redallen) [reply packages/patternfly-4/react-core/src/components/Badge/Badge.tsx]: Done for both.

File: packages/patternfly-4/react-core/src/components/Badge/index.js [hunk 2]
```diff
@@ -1 +1 @@
-export { default as Badge } from './Badge';
+yexport { default as Badge } from './Badge';
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/Badge/index.js]: Unintended change

**Comment by redallen** [comment]
Rebase on latest master to fix CircleCI errors and get a deploy preview.

**Comment by dlabaj** [comment]
@redallen Can you split these up into seperate PRs for each component.  There's to many for one Pull request to review.  Thanks.

**Comment by jschuler** [comment]
Agree with @dlabaj , this way we can also split up the work. I like the idea of having a master list of components in the issue like you have in your description though

**Comment by redallen** [comment]
Yeah, this PR is too monstrous. I'll split it up into a few smaller PRs.

**Pull request closed without merging**